### ISSUE TITLE
Optimize our pre-commit hook

### ISSUE THREAD

**Issue by jollyjerr** [CodeForBoulder/upswyng#369]
Right now our [husky](https://github.com/typicode/husky) pre-commit hook runs linting checks on basically every file before someone can make a commit. While this is rad, it takes quite a bit of time, _especialy_ if someone is using an older machine.

It would be awesome if the pre-commit hook only ran linting checks on files that have been changed as part of any given commit.

**Comment by cjlynch278** [comment]
I think this may have already been implemented. The "lint-staged" step does just that. 

I changed line 58 in the package.json file from:

"pre-commit": "lint-staged" 
to 
"pre-commit": "yarn lint" 

and found that the linter ran on the whole repo on commit. Reverting the change did not run the linter since there were not files changed. 

There may be another way to optimize this pre-commit hook, however I wasn't able to reproduce the slowness. Lemme know what you think.
